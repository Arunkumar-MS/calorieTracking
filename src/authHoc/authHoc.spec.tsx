jest.mock('js-cookie', ()=> ({get: () => 'token'}))
import { jest } from '@jest/globals'
import { shallowToJson } from "enzyme-to-json";
import * as React from 'react';
import { shallow } from 'enzyme';
import withAuth from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
    useSelector: jest.fn(),
    useDispatch: () => mockDispatch
}));

const routerMock = jest.fn();

jest.mock("swr", () => ({
    __esModule: true,
    namedExport: jest.fn(),
    default: () => ({
        data: { role: 'admin', name: 'sd', userId: '23234', calorieLimit: 123 },
    }),
}));

describe('WithAuth page', () => {
    it('should render properly', function () {
        jest.clearAllMocks();
        const Comp = withAuth(() => <div />, ['admin', 'user']);
        const wrap = shallow(<Comp user={{ role: 'admin', name: 'sd', userId: '23234', calorieLimit: 123 }} />)
        expect(shallowToJson(wrap)).toMatchSnapshot();
    })
})
