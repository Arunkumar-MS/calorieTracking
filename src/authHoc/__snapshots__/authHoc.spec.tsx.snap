// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithAuth page should render properly 1`] = `
<Layout>
  <Component
    user={
      Object {
        "calorieLimit": 123,
        "name": "sd",
        "role": "admin",
        "userId": "23234",
      }
    }
  />
</Layout>
`;
